packages-exp/auth-exp/src/core/strategies/phone.test.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+
+import { ApplicationVerifier, OperationType } from '@firebase/auth-types-exp';
+import { FirebaseError } from '@firebase/util';
+
+import { mockEndpoint } from '../../../test/api/helper';
+import { testAuth } from '../../../test/mock_auth';
+import * as fetch from '../../../test/mock_fetch';
+import { Endpoint } from '../../api';
+import { Auth } from '../../model/auth';
+import { IdTokenResponse } from '../../model/id_token';
+import { RecaptchaVerifier } from '../../platform_browser/recaptcha/recaptcha_verifier';
+import { _verifyPhoneNumber, signInWithPhoneNumber } from './phone';
+
+use(chaiAsPromised);
+use(sinonChai);
+
+describe('core/strategies/phone', () => {
+  let auth: Auth;
+  let verifier: ApplicationVerifier;
+  let sendCodeEndpoint: fetch.Route;
+
+  beforeEach(async () => {
+    auth = await testAuth();
+    auth.settings.appVerificationDisabledForTesting = true;
+    fetch.setUp();
+
+    sendCodeEndpoint = mockEndpoint(Endpoint.SEND_VERIFICATION_CODE, {
+      sessionInfo: 'session-info'
+    });
+
+    verifier = new RecaptchaVerifier(document.createElement('div'), {}, auth);
+    sinon.stub(verifier, 'verify').returns(Promise.resolve('recaptcha-token'));
+  });
+
+  afterEach(() => {
+    fetch.tearDown();
+    sinon.restore();
+  });
+
+  describe('signInWithPhoneNumber', () => {
+    it('calls verify phone number', async () => {
+      await signInWithPhoneNumber(auth, '+15105550000', verifier);
+
+      expect(sendCodeEndpoint.calls[0].request).to.eql({
+        recaptchaToken: 'recaptcha-token',
+        phoneNumber: '+15105550000'
+      });
+    });
+
+    context('ConfirmationResult', () => {
```

Comment:
can you test the callback getting called?

Reply:
That's covered in the test "calling #confirm finishes the sign in flow"

The callback calls `signInWithCredential()` which itself calls the `SIGN_IN_WITH_PHONE_NUMBER` endpoint